src/mahoji/lib/abstracted_commands/nightmareCommand.ts

```diff
@@ -109,20 +110,41 @@ async function checkReqs(
 	}
 }
 
-function perUserCost(isPhosani: boolean, quantity: number) {
+function perUserCost(user: MUser, quantity: number, isPhosani: boolean, hasShadow: boolean, hasSang: boolean) {
 	const cost = new Bank();
+	const tumCharges = shadowChargesPerKc * quantity;
+	const sangCharges = sangChargesPerKc * quantity;
+	if (hasShadow && user.user.tum_shadow_charges < tumCharges) {
```

Comment:
You are only providing a shadow or sang boost during phosani, however you are checking regardless of type fought